java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -145,25 +138,11 @@ public void reportIssue(JavaCheck javaCheck, Tree startTree, Tree endTree, Strin
 
   @Override
   public List<String> getFileLines() {
-    if(sonarComponents == null) {
```

Comment:
sonarComponents is still marked as being `@Nullable` in the constructor of the `DefaultJavaFileScannerContext`. Are you sure you can remove the null check?